### ISSUE TITLE
Would any of this belong here?

### ISSUE THREAD

**Issue by narnold0** [JuliaGraphs/Graphs.jl#267]
Hi there, 

I've been working with graph crossings and their properties and such. Originally I was using Networkx but I've been using Graphs.jl and Julia now. Some of the things I've been using are mapping from the edge pairs  `(e,e)->crossing`, if they have one that is ofc . The `# crossings per edge` (so can check for k-planarity of embedding, crossing count etc...), `crossing per vertex` (the # of crossings of edges on that vertex), sparse matrix of the crossings in the graph embedding, and also graph constructions from the crossings in a graph embedding. Would that be the kind of thing that would be welcome here? 
![Screenshot 2023-06-29 023755](https://github.com/JuliaGraphs/Graphs.jl/assets/69324584/95a8d374-515c-4f4d-8a24-17866f2fc821)

**Comment by gdalle** [comment]
Hi, thanks for offering to contribute!
The specific functions you describe would feel more at home in [Graphs.jl](https://github.com/JuliaGraphs/Graphs.jl), where all the algorithms will live in the future. GraphsBase.jl is only meant to host a basic interface and a few standard graph types

**Comment by gdalle** [comment]
Can you reopen your issue there, and maybe a PR if you want?

**Comment by narnold0** [comment]
> Can you reopen your issue there, and maybe a PR if you want?

 How do I know if I'm in the wrong one, for me it says it is in Graphs.jl here

**Comment by gdalle** [comment]
Indeed, sorry for the mistake. Then your PR is welcome here, although reviewing times may vary 😊

**Comment by etiennedeg** [comment]
Note that for the moment, this library lacks even the most basic functionalities for planarity and embeddings (which is welcome contribution, of course) although something was started here: https://github.com/JuliaGraphs/Graphs.jl/pull/208

**Comment by narnold0** [comment]
Is there a standard or consistent way of storing information related to a graph, or in relating information in Julia in general? Would this be a struct or something? methods? I only just started using Julia really and there can be a lot of ways to accomplish something sometimes so I'm not sure what I should be doing

For example, for myself, so far I've just been doing it such that when I use a function with a pair `(G,layout)` it checks a nested Dict, uses G as a key first (or creates), then uses the positions in a Dict below that as a key, my thinking being that you could be able to have multiple layouts associated with a single graph, and be able to differentiate in some way but also keep them related somehow. And that was where I was storing other dictionary mappings that were useful for functions, and it seemed faster always to just access it and use what I needed or to pass this dictionary and use what was needed.
So I was using the dictionary accessed by the pair to store key mappings and information to be accessed for the functions involved like 

  
    crossinginfo2 = crossing_construction(H, positions2)
    #accessed then with keys 
    crossinginfo2[:crossings]
    crossinginfo2[:graph]
    crossinginfo2[:positions]
    ccrossinginfo2[:edges_to_segments]
etc..
So it was easy for me to then use a function like 

     crossing_edge(info::Dict, e)

to return the edges that cross with e, and in general update/store information if I was creating it for the first time or such

**Comment by gdalle** [comment]
> Is there a standard or consistent way of storing information related to a graph, or in relating information in Julia in general?

The answer is: "not yet, but soon". We're working on a full rewrite that will allow easy manipulation of metadata associated with vertices or edges (see GraphsBase.jl). In the meantime, shortest path algorithms for example take an edge weight matrix as additional argument. So your approach of giving `layout` (which I assume contains vertex coords?) seems fine to me.
As for the output, I don't think I really understand what you're trying to do with the dictionary, but it would probably be more Julian to use a custom struct instead. If you need guidance about that I'll try to give some pointers when I review your PR
